### PR TITLE
Vault token renew based on TTL

### PR DESCRIPTION
**Pull request by lucasgolino**
This PR, will add capability to chevron to renew token after TTL expiration, avoid to call token validation always when do any operation in Vault backend reduce operations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ These are the Environment Variables that you can set to manage the webserver:
 *   `VAULT_ADDRESS` => Hashicorp Vault URL
 *   `VAULT_SKIP_VERIFY` => Hashicorp Vault Skip Verify SSL Certs on Connection
 *   `VAULT_ROOT_TOKEN` => Hashicorp Vault Root Token
+*   `VAULT_TOKEN_TTL` => Hashicorp Vault Token TTL (for exemple `24h`, for more information see https://golang.org/pkg/time/#ParseDuration)
```
Comment (racerxdl) [inline README.md]: `exemple`

**Comment by racerxdl** [comment]
Acho que vale esse fix tmb:
https://deepsource.io/gh/quan-to/chevron/run/8863f245-626c-4e5e-909a-b7f0be7e965c/go/SCC-S1008

File: vaultManager/vaultManager.go [hunk 2]
```diff
@@ -63,34 +99,58 @@ func baseVaultPath(dataType string) string {
 	return fmt.Sprintf("%s/%s/%s", remote_signer.VaultBackend, dataType, remote_signer.VaultNamespace)
 }
 
-func (vm *VaultManager) getClient() *api.Client {
+func (vm *VaultManager) validTokenTTL() bool {
+	if vm.token.getTime != nil {
+		var now = time.Now().Unix()
+		var timeWithTTL = vm.token.getTime.Unix() + int64(vm.token.ttl.Seconds())
```
Comment (rdleal) [inline vaultManager/vaultManager.go]: The value of `timeWithTTL`  my cause the usage of an expired token, as it's pointing to the exact time the token expire. A common practice for avoiding this issue is to calculate this value with some minutes before the actual expiration:
```suggestion
		var timeWithTTL = vm.token.getTime.Add(vm.token.ttl).Add(time.Minute*-1)
```

File: vaultManager/vaultManager.go [hunk 3]
```diff
@@ -63,34 +99,58 @@ func baseVaultPath(dataType string) string {
...
 
-	options := map[string]interface{}{
-		"token": vm.client.Token(),
+		if now >= timeWithTTL {
```
Comment (rdleal) [inline vaultManager/vaultManager.go]: This line could be reduced to a return statement using the [After](https://golang.org/pkg/time/#Time.After) method of the time package for indicating if the `timeWithTTL` is expired.
```suggestion
		return timeWithTTL.After(now)
```

**Pull request merged**